.githooks/pre-commit

```diff
@@ -7,8 +7,16 @@ RED='\033[0;31m'
 NC='\033[0m' # No Color
 
 # Check that Rust formatting rules are not violated.
-if ! cargo fmt -- --check; then
-    echo -e "${RED}Commit error!${NC}"
-    echo "Please format the code via 'cargo fmt', cannot commit unformatted code"
-    exit 1
-fi
+function check_fmt {
+    if ! cargo fmt -- --check; then
+        echo -e "${RED}Commit error!${NC}"
```

Comment:
Are these diffs printed to stdout? 
(if not - people might not know which directory has caused it)

Reply:
diffs are printed to stdout, yes